girder_worker/app.py

```diff
@@ -1,12 +1,136 @@
 import girder_worker
 import traceback as tb
+import celery
 from celery import Celery, __version__
 from distutils.version import LooseVersion
 from celery.signals import (task_prerun, task_postrun,
-                            task_failure, task_success, worker_ready)
+                            task_failure, task_success,
+                            worker_ready, before_task_publish)
+from celery.result import AsyncResult
 from .utils import JobStatus
 
 
+class GirderAsyncResult(AsyncResult):
+    def __init__(self, *args, **kwargs):
+        self._job = None
+        super(GirderAsyncResult, self).__init__(*args, **kwargs)
+
+    @property
+    def job(self):
+        if self._job is None:
+            try:
+                from girder.utility.model_importer import ModelImporter
```

Comment:
Could you leave a comment about how this is importing Girder only in the event that these are being called on the Girder side?